frontend/components/forms/checkbox/component.tsx

```diff
@@ -1,34 +1,52 @@
-import { FC } from 'react';
+import React, { useRef, useState } from 'react';
+
+import { useFocusRing } from 'react-aria';
+import { FieldValues } from 'react-hook-form';
 
 import cx from 'classnames';
 
-import useStatus from '../utils';
+import { useCheckbox } from '@react-aria/checkbox';
+import { useToggleState } from '@react-stately/toggle';
+
+import { CheckboxProps } from './types';
 
-import { THEME } from './constants';
-import type { CheckboxProps } from './types';
+export const Checkbox = <FormValues extends FieldValues>(props: CheckboxProps<FormValues>) => {
+  const { register, registerOptions, ...rest } = props;
+  const { name, id, 'aria-label': ariaLabel, className } = rest;
 
-export const Checkbox: FC<CheckboxProps> = ({
-  theme = 'dark',
-  input,
-  meta = {},
-  disabled = false,
-  className,
-  ...props
-}: CheckboxProps) => {
-  const st = useStatus({ meta, disabled });
+  const [invalid, setInvalid] = useState(false);
+  const state = useToggleState(props);
+  const { isSelected, toggle } = state;
+  const { isFocusVisible, focusProps } = useFocusRing(rest);
 
+  const ref = useRef();
+  const { inputProps } = useCheckbox(rest, state, ref);
+  console.log(isFocusVisible);
   return (
     <input
-      {...input}
-      {...props}
-      type="checkbox"
-      disabled={disabled}
-      className={cx({
-        'form-checkbox': true,
-        [THEME[theme].base]: true,
-        [THEME[theme].status[st]]: true,
-        [className]: !!className,
+      className={cx(
```

Comment:
You can greatly simplify the classes (have a look at the other form inputs). For example, you don't need `isSelected` or `isFocusVisible`. Use `:checked` or `:focus` instead. You also don't need to specify the duration of the animations.

Reply:
Instead of `hover:transition-all` I would prefer `transition-all` so that we always have animations. If you can also use `transition` instead of `transition-all`, it's even better.

Reply:
I don't think `bg-[url("/images/checkbox-checked.svg")]` is correct. It should be `checked:bg-[url("/images/checkbox-checked.svg")]` instead, no?

Reply:
Don't forget to add `outline-none` in the list of classes otherwise we'll keep seeing the default focus ring of the browser.